android/src/main/java/io/sentry/capacitor/SentryCapacitor.java

```diff
@@ -162,6 +163,50 @@ private Level logLevel(int level) {
         }
     }
 
+    @PluginMethod
+    public void setUser(PluginCall call) {
+        Sentry.configureScope(scope -> {
+            if (!call.getData().has("user") && !call.getData().has("otherUserKeys")) {
+                scope.setUser(null);
+            } else {
+                User userInstance = new User();
+
+                if (call.getData().has("user")) {
+                    JSObject user = call.getObject("user");
+                    if (user.has("email")) {
+                        userInstance.setEmail(user.getString("email"));
+                    }
+
+                    if (user.has("id")) {
+                        userInstance.setId(user.getString("id"));
+                    }
+
+                    if (user.has("username")) {
+                        userInstance.setUsername(user.getString("username"));
+                    }
+
+                    if (user.has("ip_address")) {
+                        userInstance.setIpAddress(user.getString("ip_address"));
+                    }
+                }
```

Comment:
Do we need an `else` here to clear out the user if nothing?

Reply:
good catch!

Reply:
@joshdholtz, actually I think the enclosing if statement should take care of this? It checks whether both the "user" and "otherUserKeys" is available, and if not, sets the scope to null.

Reply:
The enclosing if will clear if both are not set. What if only one of them is available? Is there no way to clear just one? I think that is my concern. I don't know how the other SDKs handle that 🤷‍♂️

Reply:
@joshdholtz this is how the React Native SDK was setup. I think the idea is that as long as one of the arguments ("user" or "otherUserKeys") is there, it can set that as the userInstance. Mostly likely it seems "user" will be the argument being sent and "otherUserKeys" is optional.

@bruno-garcia perhaps you can provide some insight? I'm less familiar with the "user" and "otherUserKeys" data and how they are eventually used by the Sentry API.